src/components/Button/Button.vue

```diff
@@ -0,0 +1,424 @@
+		},
+
+		iconAndText() {
+			return this.hasIcon && this.hasText
+		},
+
+		// Classes applied to the button element
+		buttonClassObject() {
+			return {
+				// If icon only, some additional css rules are required
+				'button-vue--icon-only': this.iconOnly,
+				'button-vue--text-only': this.textOnly,
+				'button-vue--icon-and-text': this.iconAndText,
+				[`button-vue--vue-${this.type}`]: this.type,
+				'button-vue--wide': this.wide,
+				'button-vue--tabbed': this.tabbed,
+			}
+		},
+	},
+
+	mounted() {
+		/**
+		 * Always fill either the text prop or the ariaLabel one.
+		 */
+		if (!(this.text && this.ariaLabel)) {
+			console.warn('You need to fill either the text or the ariaLabel props in the button component.')
+		}
+		/**
+		 * Check icon slot on first run.
+		 */
+		this.checkIconSlot()
+		/**
+		 * Register observer that checks the icon slot if programmatically changed.
+		 */
+		const observer = new MutationObserver(this.checkIconSlot)
+		observer.observe(this.$el, {
+			childList: true,
+			subtree: true,
+		})
+		this.observer = observer
+	},
+
+	beforeDestroy() {
+		this.observer.disconnect()
+	},
+
+	methods: {
+
+		/**
+		 * When the tab key is lifted, the button has been "tabbed in",
+		 * see comments on the `tabbed` variable declared in the data.
+		 */
+		handleTabUp() {
+			this.tabbed = true
+		},
+
+		/**
+		 * Everytime the button is blurred, we remove the tabbed state.
+		 */
+		handleBlur() {
+			this.tabbed = false
+		},
+
+		/**
+		 * When the button is reached via keyboard navigation and pressed using
+		 * the enter key, we slightly change the styles to provide an "active-like"
+		 * feedback. When using the mouse this is achieved with the ripple effect.
+		 */
+		makeActive() {
+			this.tabbed = false
+		},
+
+		makeInactive() {
+			this.tabbed = true
+		},
+
+		/**
+		 * Perform checks on the on what's been passed into the icon slot.
+		 */
+		checkIconSlot() {
+			// Check if the slot is populated first
+			if (Object.keys(this.$slots).length !== 0) {
+				const slottedElement = this.$slots.default[0]
+				if (slottedElement) {
+					// Check if it's a Material Design Icon
+					if (!slottedElement.data.staticClass.includes('material-design-icon')) {
+						console.warn("DESIGN WARNING: You need to use a material design icon for this button's icon. More info here: https://www.npmjs.com/package/vue-material-design-icons")
+						return
+					}
+					// Check that the MDI icon has size 20.
+					if (slottedElement.elm.clientHeight !== 20) {
+						console.warn('DESIGN WARNING: You need to set the size of the material design icon to 20px.')
+					}
+
+					// Check that the title is removed so that the tooltip on the icon is removed too.
+					if (slottedElement.data.ariaLabel !== '') {
+						console.warn('DESIGN WARNING: You need to add `title=""` as a prop on the material design icon that you passed into the default slot of the Button component.')
+					}
+				}
+			}
+		},
```

Comment:
I'm strongly against that. That sis too much.
If I want to use an inline image and not a MaterialIcon, so be it.
What if we suddenly want a bigger button for a specific area, etc etc.

The component needs to be simple, let's not over-charge this with unnecessary checks.

Reply:
Btw, if you want to enforce the icon size, you can most likely do that with css

Reply:
Ok, let's remove all this then!